libres/lib/analysis/ies/ies.cpp

```diff
@@ -167,46 +164,38 @@ void ies_initX__(const matrix_type *A, const matrix_type *Y0,
      * CONSTRUCT TRANFORM MATRIX X FOR CURRENT ITERATION (Line 10)
      *   X= I + W/sqrt(N-1)
      */
-    matrix_assign(X, W0);
-    matrix_scale(X, nsc);
+    X = W0;
+    X *= nsc;
     for (int i = 0; i < ens_size; i++) {
-        matrix_iadd(X, i, i, 1.0);
+        X(i, i) += 1.0;
```

Comment:
This I think can be replaced with `X.diagonal().array() += 1`.

I tested this and it works:

```cpp
  Eigen::MatrixXd A{{1, 0}, {0, 1}, {0, 0}};
  A.diagonal().array() += 1;
```